### PR TITLE
feature: simple support of create pull request

### PR DESCRIPTION
**Pull request by randy3k**
The first attempt to provide "Create Pull Request" feature in #318. There are still a lot to do, for example,

1. check upstream is set and updated
1. allow pr to different branches and remotes.

PS: This PR was launched by  using `github: create pull request` 😊

### CODE REVIEW

**Comment by randy3k** [comment]
Oh, just notice a problem: "remote name" may not equal to "git user name"...

File: core/git_mixins/remotes.py [hunk 1]
```diff
@@ -58,4 +58,17 @@ def get_upstream_for_active_branch(self):
...
-        return self.git("rev-parse", "--abbrev-ref", "--symbolic-full-name", "@{u}", throw_on_stderr=False)
+        return self.git("rev-parse", "--abbrev-ref", "--symbolic-full-name",
+                        "@{u}", throw_on_stderr=False).strip()
+
+    def get_active_remote_branch(self):
+        """
+        Return named tuple of the upstream for active branch.
+        """
+        upstream = self.get_upstream_for_active_branch()
+        selected_remote_branch = None
```
Comment (divmain) [inline core/git_mixins/remotes.py]: Prefer the form:
```python
for branch in self.get_branches():
    if branch.name_with_remote == upstream:
        return branch
else:
    return None
```

File: github/commands/pull_request.py [hunk 2]
```diff
@@ -131,3 +131,25 @@ def view_diff_for_pr(self):
...
+    """
+    Create pull request of the current commit on the current repo.
+    """
+
+    def run(self, edit):
+        sublime.set_timeout_async(self.run_async, 0)
+
+    def run_async(self):
+        base_remote = github.parse_remote(self.get_integrated_remote_url())
+        remote_branch = self.get_active_remote_branch()
```
Comment (divmain) [inline github/commands/pull_request.py]: There should probably be a check here to abort if there was an issue identifying the remote branch. i.e.
```python
if not remote_branch:
    sublime.message_dialog("Unable to determine remote.")
```
We can swap this out for more intelligent behavior later.

**Comment by divmain** [comment]
A couple of small things.  Let me know, and I'll merge.  Thanks!

**Comment by randy3k** [comment]
Code rebased.

**Comment by divmain** [comment]
Looks good to me.  Thanks again!

**Pull request merged**